src/lib.rs

```diff
@@ -31,6 +31,18 @@ pub trait Validate {
     }
 }
 
+impl Method {
+    pub fn get_pact<'a>(&'a self) -> Option<Contract<'a>> {
+        match &self.constraints {
+            Some(constraints) => match Contract::decode(constraints) {
+                Ok(contract) => Some(contract),
+                Err(_) => None,
```

Comment:
Can leave a comment "this error case can only occur after initializing a Method with bad constraints. A decoded Method will be checked during decoding"